build.cake

```diff
@@ -159,46 +164,69 @@ Task("BuildCppTestFiles")
 //////////////////////////////////////////////////////////////////////
 // TEST
 //////////////////////////////////////////////////////////////////////
+
+Task("CheckForError")
+    .Does(() => 
+    {
+    CheckForError(ref ErrorDetail);
+    });
```

Comment:
Very minor, but indenting is off here. Also I think you could simplify this and other short one liners down to `.Does(() => CheckForError(ref ErrorDetail));`